pandas/core/frame.py

```diff
@@ -5391,10 +5391,16 @@ def combine(self, other, func, fill_value=None, overwrite=True):
         2  NaN  3.0 1.0
         """
         other_idxlen = len(other.index)  # save for compare
+        other_columns = other.columns
 
         this, other = self.align(other, copy=False)
         new_index = this.index
 
+        if other.empty and self.empty:
+            empty_columns = self.columns.union(other_columns)
+            self = self.reindex(columns=empty_columns, index=[None]).dropna()
```

Comment:
just 
```return self.reindex(....)```

why do we need the .dropna()?

don't need to copy (as reindex will copy)

Reply:
okay, will do return instead of copy and doing .dropna() because of ` index=[None]`, it adds a single row to a dataframe.